### ISSUE TITLE
Unpredictable output path for generated files

### ISSUE THREAD

**Issue by emartinfigma** [dtolnay/cxx#949]
We are trying to integrate cxx into a larger CMake-based build system.

Following some other examples we have a `build.rs` that contains:

```
fn main() {
    let _ = cxx_build::bridge("src/lib.rs");
}
```

Ultimately cmake generates a command line that looks like:

```
cargo build --target-dir /.../build/emscripten-1.40.1/rust --target=wasm32-unknown-emscripten --release
```

What's super confusing here is that, when executed by Ninja (which prints the command line it's running), that ends up writing `lib.rs.cc`/`.h` under

`build/emscripten-1.40.1/rust/cxxbridge/...`

but if I copy paste the same command from the command line, it puts them under

`build/emscripten-1.40.1/rust/wasm32-unknown-emscripten/cxxbridge/...`

In both cases, they contain a directory tree like

```
├── cxxbridge
│   ├── fullscreen-link (our crate name)
│   │   └── src
│   │       ├── lib.rs.cc -> ...symlink to a file in the build output that has a hash in the name...
```

For the life of me I cannot figure out what is the difference between how these two commands are run.  It must be obeying some environment variable or something?  Could the names of the path components leading up to the input or output dir matter?  What else can I investigate to try figure out the cause of this?

**Comment by dtolnay** [comment]
I don't know of anything in this crate which would behave that way. I bet it's Cargo doing it. Notice the following:

```console
$ cargo new --lib repro
     Created library `repro` package

$ cd repro
$ cargo build
   Compiling repro v0.1.0
    Finished dev [unoptimized + debuginfo] target(s) in 0.19s

$ find target -name librepro.rlib
target/debug/librepro.rlib

$ cargo clean
$ cargo build --target=wasm32-unknown-emscripten
   Compiling repro v0.1.0
    Finished dev [unoptimized + debuginfo] target(s) in 0.10s

$ find target -name librepro.rlib
target/wasm32-unknown-emscripten/debug/librepro.rlib
```

which is a very similar pattern.

**Comment by emartinfigma** [comment]
Yes, the `--target` flag reliably puts the Rust outputs in the `wasm32-unknown-emscripten` subdir for me.  The real mystery here is how the cxx-generated outputs `lib.rs.cc` etc. seem to either be in that subdir or not for reasons I cannot gauge in my execution environment.  

Digging more I found the discussion here maybe relevant: https://github.com/rust-lang/cargo/issues/6100 , though perhaps the out dir is dir with hashes mentioned above.

I filed this bug in the hope you'd recognize it and say "oh yes that is X", but if you don't, then I agree this is likely some cargo thing out of your hands, so feel free to close.

I think to work around it on our side we could try using cxxbridge-cmd instead, does that make sense to you?
